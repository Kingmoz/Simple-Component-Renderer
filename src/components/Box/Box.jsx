import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

const Wrapper = styled.div`
  padding: 15px;

  background: #fff;
  border: 3px solid #fff;
  border-radius: 15px;

  ${props => props.css ? props.css : null}
`;

const ContentWrapper = styled.div`
  padding: 0 15px;
`;

const Box = ({ css, content, children }) => {
  return (
    <Wrapper css={css}>
      { content && <h4>{ content }</h4> }
      <ContentWrapper>
        { children }
      </ContentWrapper>
    </Wrapper>
  );
};

Box.propTypes = {
  css: PropTypes.object,
  content: PropTypes.string,
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ])
}

export default Box;