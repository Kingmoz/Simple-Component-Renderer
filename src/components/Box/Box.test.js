import React from 'react';
import { mount } from 'enzyme';

import Box from './Box';

function setup() {
  const props = {
    content: 'message'
  };

  return mount(<Box {...props} />);
}

describe('Box', () => {
  const wrapper = setup();

  it('should render properly', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should render content correctly', () => {
    expect(wrapper.text()).toEqual('message');
  });

});