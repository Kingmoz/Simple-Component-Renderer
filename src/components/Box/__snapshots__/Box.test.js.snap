// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box should render properly 1`] = `"<div class=\\"sc-bdVaJa kQfuFR\\"><h4>message</h4><div class=\\"sc-bwzfXH gbwKQg\\"></div></div>"`;
